docs/timezone.md

```diff
@@ -133,12 +135,85 @@ tz2 = Temporal.TimeZone.from(tz);
 ### timeZone.**id** : string
 
 The `id` property gives an unambiguous identifier for the time zone.
-Effectively, this is the canonicalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
+This is the normalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
 
 When subclassing `Temporal.TimeZone`, this property must be overridden to provide an identifier for the custom time zone.
 
 ## Methods
 
+### timeZone.**equals**(_other_: Temporal.TimeZone | object | string) : boolean
+
+**Parameters:**
+
+- `other` (`Temporal.TimeZone` object, object implementing the `Temporal.TimeZone` protocol, or a string time zone identifier): Another time zone to compare.
+
+**Returns:** `true` if `timeZone` and `other` are equal, or `false` if not.
+
+Compares two time zones for equality.
+Equality is determined by the following algorithm:
+
+- If `timeZone === other`, then the time zones are equal.
+- Otherwise, `timeZone.id` is compared to `other` (or `other.id` if `other` is an object).
+  If any of the following conditions are true, then the time zones are equal:
+  - Both string identifiers are Zone or Link names in the [IANA Time Zone Database](https://www.iana.org/time-zones), and they resolve to the same Zone name.
+    This resolution is case-insensitive.
+  - Both string identifiers are custom time zone identifiers that are equal according to `===`.
+    This comparison is case-sensitive and does not normalize Unicode characters.
+  - Both identifiers are numeric offset time zone identifiers like "+05:30", and they represent the same offset.
+- Otherwise, the time zones are not equal.
```

Comment:
By no means necessary, but this explanation can be rephrased:
```suggestion
Compares two time zones for equality.
They are reported equal if and only if both are the same object (`timeZone === other`) or a string comparison of their identifiers (`timeZone.id` against `other.id` if `other` is an object, otherwise `timeZone.id` against `other`) is described by any of the following conditions:
  - Both string identifiers are Zone or Link names in the [IANA Time Zone Database](https://www.iana.org/time-zones), and they resolve to the same Zone name.
    This resolution is case-insensitive.
  - Both string identifiers are custom time zone identifiers that are equal according to `===`.
    This comparison is case-sensitive and does not normalize Unicode characters.
  - Both identifiers are numeric offset time zone identifiers like "+05:30", and they represent the same offset.
```

Reply:
For the spec, longer sentences seem to be preferred, but for the docs I like to keep sentences shorter so it's easier for readers to follow along. So I'd prefer to leave the simpler text as-is, if that's OK, or use some other set of shorter sentences.